cincinnati/src/lib.rs

```diff
@@ -188,10 +200,15 @@ impl Graph {
             Some(id) => {
                 let node = self.dag.node_weight_mut(id.0).expect(EXPECT_NODE_WEIGHT);
                 if let Release::Concrete(_) = node {
-                    bail!(
-                        "Concrete release with the same version ({}) already exists",
-                        release.version()
-                    );
+                    // check if release digest and node digest are same
+                    if release.manifestref().unwrap() != node.manifestref().unwrap() {
+                        bail!(
+                            "mismatched manifest ref for concrete release {}:{}, {}",
```

Comment:
nit: missing a space after the colon?  `mismatched manifest ref for concrete release {}: {}, {}`